File: AzW.Web.Portal/src/models/AppGateway.js
import ResourceType from './ResourceType'
export default class AppGateway
{
    constructor (){
        this.GraphModel = {
            Id: '',
            ResourceType: ResourceType.AppGw(),
            DisplayName: ''
        };

        this.ProvisionContext = {
            ResourceType: ResourceType.AppGw(),
            Name: '',
            Location: '',
            ResourceGroupName: '',
            VNetName: '',
            SubnetName: '',
            NumberofInstances: 2,
            WAFEnabled: false,
            RoutingRuleName: 'Rule-HTTP-80-to-80',
            FrontendListenerName: 'FEListener-Http-80',
            FrontendPort: 80,
            BackendPoolName: 'Backendpool-VMs',
            BackendPort: 80,
            LoadBalanceToExistingVMNames: ''
        };
        this.CalculatorContext = {
            IsFree: false,
            Tier: ''
        }
    }
}

File: src/ApiServer/AzW.Model/Provision/AppGateway.cs

using System.Collections.Generic;

namespace AzW.Model
 {
     public class AppGateway : Resource
     {
          public string Location { get; set; }
          public string ResourceGroupName { get; set; }
          public string VNetName { get; set; }
          public string SubnetName { get; set; }
          public int NumberofInstances { get; set; } = 1;
          public bool WAFEnabled { get; set; } = false;
          public string FrontendListenerName { get; set; }
          public string RoutingRuleName { get; set; }
          public int FrontendPort { get; set; }
          public int BackendPort { get; set; }
          public string BackendPoolName { get; set; }
          public string[] LoadBalanceToExistingVMNames { get; set; }
     }
 }